runtime/domain.c

```diff
@@ -1147,114 +1146,147 @@ CAMLexport _Atomic caml_timing_hook caml_domain_terminated_hook =
 
 static void domain_terminate(void);
 
-static value make_finished(caml_result result)
+static value make_finished(value res_or_exn)
```

Comment:
Wait, this change is moving code from the new/better API to the old/worse API. This is the wrong direction, right?